File: python3/1310/Solution.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        answer = []

        xors = []
        xors.append(arr[0])

        for index in range(1, len(arr)):
            xors.append(xors[-1] ^ arr[index])

        for query in queries:
            start_index = query[0]
            end_index = query[1]

            answer.append(arr[start_index] ^ xors[start_index] ^ xors[end_index])

        return answer

File: java/1310/Solution.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int arrayCount = arr.length;
        int queryCount = queries.length;

        int[] answer = new int[queryCount];
        int[] xors = new int[arrayCount];

        xors[0] = arr[0];

        for (int arrayIndex = 1; arrayIndex < arrayCount; arrayIndex++) {
            xors[arrayIndex] = xors[arrayIndex - 1] ^ arr[arrayIndex];
        }

        for (int queryIndex = 0; queryIndex < queryCount; queryIndex++) {
            int[] query = queries[queryIndex];

            int startIndex = query[0];
            int endIndex = query[1];

            answer[queryIndex] = arr[startIndex] ^ xors[startIndex] ^ xors[endIndex];
        }

        return answer;
    }
}
